### ISSUE TITLE
Cannot import open3D that I build in windows

### ISSUE THREAD

**Issue by dlwns97** [isl-org/Open3D#6273]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

#### I first cloned Open3D by (I did't move to Open3D by cd):
```
git clone https://github.com/isl-org/Open3D.git
```

#### Then, I build Open3D (on windows x64, with no CUDA (I failed to build with cuda)) with:
```
mkdir build
cd build
cmake -DPYTHON_EXECUTABLE=<python path> -S source_dir -B build_dir
cmake --build . --config Release --target INSTALL

```
#### Then, I activate my virtual env with:
```
python -m venv myenv
myenv\Scripts\Activate.ps1 (with powershell)

```
#### Install packages with:
```
cmake --build . --config Release --target install-pip-package
cmake --build . --config Release --target python-package
cmake --build . --config Release --target pip-package
```

#### Finally I did install open3d with:
```
pip install mydirectory\build\lib\python_package\pip_package\open3d-0.17.0+6ddbcd5c9-cp39-cp39-win_amd64.whl
```
### Error message

```shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mydirectory\AppData\Local\Programs\Python\Python39\lib\site-packages\open3d\__init__.py", line 93, in <module>
    from open3d.cpu.pybind import (core, camera, data, geometry, io, pipelines,
ImportError: RedwoodIndoorLivingRoom1: PyType_Ready failed (UnicodeDecodeError: 'utf-8' codec can't decode byte 0x9c in position 262: invalid start byte)!
```

### Open3D, Python and System information

```markdown
- Operating system: Windows 11 64-bit
- Python version: Python 3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]
- Open3D version: 0.17.0+6ddbcd5c9 (I can't see open3D version with python but I got result by pip list)
- System architecture: arm64
- Is this a remote workstation?: no, I did on windows powershell
- How did you install Open3D?: pip build from source
- Compiler version (if built from source): cmake version 3.27.0 (I don't have idea about compiler sorry)
```

### Additional information

![image](https://github.com/isl-org/Open3D/assets/58970803/a59c24ec-7db0-4da4-8b1c-535cf4395016)

![image](https://github.com/isl-org/Open3D/assets/58970803/b0da5c4f-d54a-4481-b5ed-648adab2e311)

**Comment by dlwns97** [comment]
I've done same way with your link but still I got same error,
![image](https://github.com/isl-org/Open3D/assets/58970803/7938d8fb-6b00-44aa-82b0-80c93488935a)

I replace original source file and build.

1. Replace cpp/pybind/data/dataset.cpp
2. cmake -DPYTHON_EXECUTABLE=<python path> -S source_dir -B build_dir
3. cmake --build . --config Release --target ALL_BUILD
4. cmake --build . --config Release --target INSTALL
5. (Activate virtual env) cmake --build . --config Release --target install-pip-package
6. cmake --build . --config Release --target python-package
7. cmake --build . --config Release --target pip-package
8. And pip install .whl file in virtual env 
9. But still got same error.... 

Do you have any recommend way?

**Comment by dlwns97** [comment]
I think we should also change Open3D/cpp/open3d/data/Dataset.h 
![image](https://github.com/isl-org/Open3D/assets/58970803/0537085e-6e0c-4a56-9d2c-5eec94450ee6)

Dataset.h also has some wrong characters that you show in https://github.com/isl-org/Open3D/pull/6277

**Comment by ssheorey** [comment]
This is fixed in master with the recent PR #6277 from @MisakaQwQ
